### ISSUE TITLE
Add part attributes to docs-description content

### ISSUE THREAD

**Issue by christophe-g** [web-padawan/api-viewer-element#20]
Let's say I would like to style `h2` from a markdown description like: 

```md
 ## My new element

describe my element. 
```

I am aware I can use `::part(docs-description)` to set inherited css properties, like `font-family`. For instance, this will work (and all child elements from `[part=doc-description]` will inherit from the style): 

``` css
      api-viewer::part(docs-description) {
        font-family: 'Roboto', 'Noto', sans-serif;
        -webkit-font-smoothing: antialiased;  /* OS X subpixel AA bleed bug */
      }
```

However, what would be the option if I want to style a `h2` element within my markdown description? This for instance has no effect on h2 title element: 

```css
api-viewer::part(docs-description) h2 {
    font-size: 34px;
    font-weight: 400;
    letter-spacing: -.01em;
    line-height: 40px;
}
```

Any other better alternative than dynamically appending a `link[rel=stylesheet]` tag to `api-viewer-content` ?

**Comment by web-padawan** [comment]
Hi, if you want to provide your own styles, you can import [base class](https://github.com/web-padawan/api-viewer-element/blob/master/src/api-viewer-base.ts) and then register your own custom element, which lets you provide your own styles using `css` tagged literal.

However I think we can fix that by doing a set of RegExp replacements to `marked` output:

```js
'<h2>'.replace(/<(h([2-6]))>/, '<$1 part="docs-$1">'); // "<h2 part="docs-h2">
```

So, thanks for the idea! I think this improvement makes a lot of sense. Will add to my TODO list.

**Comment by christophe-g** [comment]
Thanks for this rapid answer !

> Hi, if you want to provide your own styles, you can import base class and then register your own custom element, which lets you provide your own styles using css tagged literal.

That makes sense. So we would have - in the extended custom element

```js
static get styles() {
    return [
      ...super.styles,
     css`
     [part='docs-description'] h2 {
        your new style
      }
     `
  }
 
```

As we can already provide templates for demo, what about using a similar mechanism for injecting style (I suspect the proposed regex solution might fall short in a lot of situation - what about table, list...)? 

```html
<template data-target="parent"> 
   <style> 
      your style
   </style>
</template>
```

I miss `vaadin-theme` sometimes ; )

**Comment by web-padawan** [comment]
I'm ok to also provide API for injecting styles, but please create another issue where we could discuss it so you could better illustrate the problem that you are trying to solve.

Regarding Vaadin styling approach, the ultimate idea of `<api-viewer>` is to avoid any opinionated solutions and to purely rely on the platform API - for example, `<template>` and `::part`.

**Comment by christophe-g** [comment]
@web-padawan - all right and many thanks. 

Actually, extending the base class works well and solves my use case. 

For reference and potential future users having similar requirements : 

``` js

import { ApiViewer } from 'api-viewer-element';
import { css } from 'lit-element';

class DemoApiViewer extends ApiViewer {

  static get styles() {
    return [
      super.styles,
      css `
      
       [part='docs-description'] h1 {
          font-size: 45px;
          font-weight: 400;
          letter-spacing: -.018em;
          line-height: 48px;
        }

        [part='docs-description'] h2 {
          font-size: 34px;
          font-weight: 400;
          letter-spacing: -.01em;
          line-height: 40px;
        }

        [part='docs-description'] h3 {
          font-size: 24px;
          font-weight: 400;
          letter-spacing: -.012em;
          line-height: 32px;
        }

        [part='docs-description'] h4 {
          font-size: 16px;
          font-weight: 400;
          line-height: 24px;
        }

        [part='docs-description'] h5, [part='docs-description'] h6 {
          font-size: 14px;
          font-weight: 500;
          line-height: 24px;
        }
   `
    ];
  }
}

customElements.define('demo-api-viewer', DemoApiViewer);


```

**Comment by web-padawan** [comment]
Done by #22 and will be included to the upcoming `0.4.0` release.
